docs/standard/garbage-collection/implementing-dispose.md

```diff
@@ -1,158 +1,173 @@
-  
-  **Managed objects that consume large amounts of memory or consume scarce resources.** Freeing these objects explicitly in the `Dispose` method releases them faster than if they were reclaimed non-deterministically by the garbage collector.  
-  
-If the method call comes from a finalizer (that is, if *disposing* is `false`), only the code that frees unmanaged resources executes. Because the order in which the garbage collector destroys managed objects during finalization is not defined, calling this `Dispose` overload with a value of `false` prevents the finalizer from trying to release managed resources that may have already been reclaimed.  
-  
+[!code-vb[Conceptual.Disposable#7](../../../samples/snippets/visualbasic/VS_Snippets_CLR/conceptual.disposable/vb/dispose1.vb#7)]
+
+The `Dispose` method performs all object cleanup, so the garbage collector no longer needs to call the objects' <xref:System.Object.Finalize%2A?displayProperty=nameWithType> override. Therefore, the call to the <xref:System.GC.SuppressFinalize%2A> method prevents the garbage collector from running the finalizer. If the type has no finalizer, the call to <xref:System.GC.SuppressFinalize%2A?displayProperty=nameWithType> has no effect. Note that the actual work of releasing unmanaged resources is performed by the overload of the `Dispose` method.
+
+### The Dispose(bool) method overload
+
+In the overload, the `disposing` parameter is a <xref:System.Boolean> that indicates whether the method call comes from a <xref:System.IDisposable.Dispose%2A> method (its value is `true`) or from a finalizer (its value is `false`).
+
+The body of the method consists of two blocks of code:
+
+- A block that frees unmanaged resources. This block executes regardless of the value of the `disposing` parameter.
+
+- A conditional block that frees managed resources. This block executes if the value of `disposing` is `true`. The managed resources that it frees can include:
+
+  **Managed objects that implement <xref:System.IDisposable>.** The conditional block can be used to call their <xref:System.IDisposable.Dispose%2A> implementation. If you have used a safe handle to wrap your unmanaged resource, you should call the <xref:System.Runtime.InteropServices.SafeHandle.Dispose%28System.Boolean%29?displayProperty=nameWithType> implementation here.
+
+  **Managed objects that consume large amounts of memory or consume scarce resources.** Freeing these objects explicitly in the `Dispose` method releases them faster than if they were reclaimed non-deterministically by the garbage collector.
+
+If the method call comes from a finalizer (that is, if `disposing` is `false`), only the code that frees unmanaged resources executes. Because the order in which the garbage collector destroys managed objects during finalization is not defined, calling this `Dispose` overload with a value of `false` prevents the finalizer from trying to release managed resources that may have already been reclaimed.
```

Comment:
Calling Dispose with false doesn't prevent things.  The burden is on the implementer to be careful to make sure that's the false path doesn't interact with managed objects that may have potentially been reclaimed.

Knowing that an object has not potentially been reclaimed is an exercise left to the reader :smile:.

Reply:
For the purpose of this article, we keep describing the code examples for what they're doing and why - rather than what the dispose pattern, specifically the implementation of the `Dispose(bool)` is intended to do. Should we pivot a bit on this, and make it more declarative - for example, state the intent for managed vs unmanaged resources and how the `disposing` parameter should be used with each?

Reply:
>  Should we pivot a bit on this, and make it more declarative 

That's already the tone I thought it was going for, because it's prose before sample (example for clarity) vs sample then prose (example for analysis).